ecosystem/sep-0043.md

```diff
@@ -0,0 +1,102 @@
+## Preamble
+
+```
+SEP: 0043
+Title: Standard Web Wallet API Interface
+Authors: Piyal Basu <piyal@stellar.org>, Leigh McCulloch <@leighmcculloch>, George Kudrayvtsev <george@stellar.org>, Enrique Arrieta <@earrietadev>, Orbit Lens <@orbitlens>
+Track: Standard
+Status: Draft
+Created: 2024-04-11
+Version: 1.0.0
+Discussion: https://github.com/stellar/stellar-protocol/discussions/1467
+```
+
+## Abstract
+
+This standard aims to enhance developer experiences by standardizing an interface for web wallet's to use
+when interacting with other codebases.
+
+## Motivation
+
+There are numerous Stellar wallets in the ecosystem with this number expected to grow as the ecosystem does.
+Wallets expose an API to allow dapps to call methods that preform such common tasks as transmitting a wallet's adress. An issue is that wallets all use slightly different API's to preform similar functions, with different method names and call signature as well as different return values. Developers have a motivation to integrate with as many as possible, but currently, they must create abstractions to handle every wallet they wish to connect to. This proposal presents a standard API interface for all wallets to implement. If all wallets have the same interface, it removes the needs for developers to create abstractions for every individual wallet they want to integrate with. Developers can safely assume a wallet uses this agreed upon interface, creating simpler codepaths that easily connect to current wallets while also simplifying the onboarding of new wallets.
+
+## Specification
+
+The standard web wallet interface provides a structure for common methods that wallets are expected to offer. In addition, the interface defines common error message shapes to ensure wallets adhere to informative and usable error messaging. Messages are designed to be surfaced directly to the user, so they should provide a readable description of what went wrong.
+
+### Wallet Interface Format
+
+Wallets will employ an API using the following format:
+
+```typescript
+
+interface Error {
+  message: string,    // general description message returned to the client app
+  code: number,       // unique error code
+  ext?: Array<string>  // optional extended details
+}
+
+{
+  getAddress: () => Promise<{address: string;} | Error>,
+  signTransaction: (xdr: string, opts?: { network?: string, networkPassphrase?: string, address?: string }) => Promise<{signedXdr: string} | Error>,
+  signAuthEntry: (authEntryXdr: string, opts?: { network?: string, networkPassphrase?: string, address?: string }) => Promise<{signedAuthEntry: string} | Error>,
+  signMessage: (message: string, opts?: { network?: string, networkPassphrase?: string, address?: string }) => Promise<{signedMessage: string} | Error>,
```

Comment:
:bulb: For all of these, the response should just be `Promise<{ signature: string } | Error>`. It's a better reflection of the return value, which is just an opaque signature. Also, we should describe its format: is it hex, base64, etc.? For example, [`Keypair.sign`](https://stellar.github.io/js-stellar-sdk/Keypair.html#sign) just returns raw bytes.

Unless this is implying that you would get e.g. an auth entry that is "filled out" with the signature field inside of it? This isn't explicit in the description below.

Reply:
@Shaptic This is a good call out for `signAuthEntry` and `signMessage`. 

`signTransaction` actually returns the signed transaction xdr from Freighter, not just the signautre, so I do think `Promise<{ xdr: string } | Error>` is still correct here. Though it may be useful to also pass just the signature

Reply:
Good question on the Buffer encoding piece. I know Albedo encodes as hex. I'll check with the ecosystem to see if they have a preference here